Video-Transformers/Video-Transformers/ViewController.m

```diff
@@ -0,0 +1,340 @@
+
+    // Calculate the desired size of the image
+    CGFloat desiredWidth = videoWidth / 8;  // Adjust this value as needed
+    CGFloat desiredHeight = image.size.height * (desiredWidth / image.size.width);
+
+    // Resize the image to the desired size
+    UIImage *resizedImage = [self resizeImage:image toSize:CGSizeMake(desiredWidth, desiredHeight)];
+
+    // Get pointer to the Y plane
+    uint8_t* yPlane = [videoFrame getPlaneBinaryData:0];
+    
+    // Create a CGContext from the Y plane
+    CGContextRef context = CGBitmapContextCreate(yPlane, videoWidth, videoHeight, 8, videoWidth, CGColorSpaceCreateDeviceGray(), kCGImageAlphaNone);
+    
+    // Location of the image (in this case right bottom corner)
+    CGFloat x = videoWidth * 4/5;
+    CGFloat y = videoHeight * 1/5;
+    
+    // Draw the resized image on top of the Y plane
+    CGRect rect = CGRectMake(x, y, desiredWidth, desiredHeight);
+    CGContextDrawImage(context, rect, resizedImage.CGImage);
+    
+    CGContextRelease(context);
+}
+
+@end
+
+customTransformer* logoTransformer;
+
+@implementation ViewController
+
+#pragma mark - View lifecycle
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    // Step 1: As the view comes into the foreground, initialize a new instance
+    // of OTSession and begin the connection process.
+    _session = [[OTSession alloc] initWithApiKey:kApiKey
+                                       sessionId:kSessionId
+                                        delegate:self];
+    [self doConnect];
+}
+
+- (BOOL)prefersStatusBarHidden
+{
+    return YES;
+}
+
+- (BOOL)shouldAutorotate {
+    return UIUserInterfaceIdiomPhone != [[UIDevice currentDevice] userInterfaceIdiom];
+}
+#pragma mark - OpenTok methods
+
+/** 
+ * Asynchronously begins the session connect process. Some time later, we will
+ * expect a delegate method to call us back with the results of this action.
+ */
+- (void)doConnect
+{
+    OTError *error = nil;
+    
+    [_session connectWithToken:kToken error:&error];
+    if (error)
+    {
+        [self showAlert:[error localizedDescription]];
+    }
+}
+
+/**
+ * Sets up an instance of OTPublisher to use with this session. OTPubilsher
+ * binds to the device camera and microphone, and will provide A/V streams
+ * to the OpenTok session.
+ */
+- (void)doPublish
+{
+    OTPublisherSettings *settings = [[OTPublisherSettings alloc] init];
+    settings.name = [UIDevice currentDevice].name;
+    _publisher = [[OTPublisher alloc] initWithDelegate:self settings:settings];
+   
+    OTError *error = nil;
+    [_session publish:_publisher error:&error];
+    if (error)
+    {
+        [self showAlert:[error localizedDescription]];
+    }
+    
+    [self.view addSubview:_publisher.view];
+    [_publisher.view setFrame:CGRectMake(0, 0, widgetWidth, widgetHeight)];
+
+    
+    NSString *resourcePath = [[NSBundle mainBundle] pathForResource:@"selfie_segmentation" ofType:@"tflite" inDirectory:@"OpenTok.bundle"];
```

Comment:
This should be done once.. in init maybe?

Reply:
yes, we can do that